src/core/qgsrendercontext.h

```diff
@@ -113,6 +114,12 @@ class CORE_EXPORT QgsRenderContext
      */
     QgsCoordinateTransform coordinateTransform() const {return mCoordTransform;}
 
+    /**
+     * A general purpose distance and area calculator, capable of performing ellipsoid based calculations.
+     * \since QGIS 3.0
+     */
+    QgsDistanceArea distanceArea() {return mDistanceArea;}
```

Comment:
actually it's probably best to change this to

    const QgsDistanceArea& distanceArea() const { return mDistanceArea; }